### ISSUE TITLE
Не отображаются подсказки для кнопок редактора

### ISSUE THREAD

**Issue by inliquid** [altocms/altocms#292]
Исходная проблема была решена лишь наполовину. По факту для моего случая она не решена совсем, поскольку я пользуюсь доп. плагинами. В частности я использую плагин emoji от ЛС с плагином совместимости. В результате - никаких подсказок для кнопок у меня нет.

Исходный баг -> https://github.com/altocms/altocms/issues/248

**Comment by inliquid** [comment]
А почему проблема не движка?
Движок позволяет устанавливать доп. плагины, в т.ч. от ЛС? Позволяет.
Должна ли происходить деградация основных функций при установке плагинов - очевидно нет.
Что будет если я решу установить какой-то другой?

Я не согласен с таким решением. Как workaround сойдет, но баг нужно закрывать по-другому.

**Comment by inliquid** [comment]
опять началась мистика
   'panel_title_h4'                            => 'большой заголовок H4',
    'panel_title_h5'                            => 'средний заголовок H5',
    'panel_title_h6'                            => 'малый заголовок H6',

- не срабатывает.
Некоторые другие текстовки подхватываются, например 
  'panel_user'                                => 'вставить ссылку на пользователя',

А вот
'panel_video'                               => 'видео (youtube, vimeo, rutube)',

- срабатывает ТОЛЬКО под виндой в Open Server 
O_O
Под линуксом показывает 'видео' (при том что другие текстовки поменялись)

**Comment by inliquid** [comment]
Так с видео разобрался, остались только H4-H6

**Comment by aVadim483** [comment]
При установке ЛС-плагинов деградация функционала происходить, конечно, не должна, но может, т.к. гарантировать 100%-ую совместимость с плагинами устаревшего движка вряд ли можно.

**Comment by andrey-v** [comment]
Проблема не движка потому, что 
1. В самом плагине ошибка и даже в LS при его установке подсказки исчезают  :confused:
2. А вот если использовать такой код
```js
var newMarkitupSettings = ls.settings.getMarkitup;
var newMarkitupSettingsComment = ls.settings.getMarkitupComment;

ls.settings.getMarkitup = function() {
    newMarkitupSettings = newMarkitupSettings.call(ls.settings);
    newMarkitupSettings.markupSet = newMarkitupSettings.markupSet.concat(myMarkupSet);
    return newMarkitupSettings;
};

ls.settings.getMarkitupComment = function() {
    newMarkitupSettingsComment = newMarkitupSettings.call(ls.settings);
    newMarkitupSettingsComment.markupSet = newMarkitupSettingsComment.markupSet.concat(myMarkupSet);
    return newMarkitupSettingsComment;
};
```

То плагин станет корректно работать и в Alto и в LS. Потому проблема в некорректном поведении плагина, а не в логике добавления кнопки в редактор markitup для Alto, вот.

**Comment by inliquid** [comment]
Так вроде речь шла что любой плагин, который добавляет кнопки в markitup сломает вывод текстовых подписей? Андрей?
И что с H1-H4, почему не задаются подсказки?

**Comment by andrey-v** [comment]
Да, любой плагин не сможет.
А h4-h6 - это и есть подсказки, они и выводятся. [Вот](https://github.com/altocms/altocms/blob/master/common/templates/frontend/libs/js/engine/settings.js#L29-L31) - значение **name** выводится в качестве подсказки.

**Comment by inliquid** [comment]
Но в языковом файле задаются текстовки

    'panel_title_h4'                            => 'большой заголовок H4',
    'panel_title_h5'                            => 'средний заголовок H5',
    'panel_title_h6'                            => 'малый заголовок H6',

Стало быть, косяк?

**Comment by andrey-v** [comment]
Да, действительно, то что текстовки жестко заданы в js-скрипте не хорошо. Исправил 7a7ad29b1df5ebb471f4c45316b09216a5de80aa

**Comment by lexuspro** [comment]
это только для шаблона Experience? В start-kit не наблюдаю.

**Comment by andrey-v** [comment]
Для обоих тем, но в языковом файле написано:
**'panel_title_h4'  => 'H4'**, а не **'большой заголовок H4'** потому выводится **H4** и ничего не заметно.
